mscr-ui/src/modules/crosswalk-editor/mappings-accordion/index.tsx

```diff
@@ -239,22 +253,31 @@ function Row(props: {
                   </Sbutton>
                 </Tooltip>
                 <Tooltip
-                  title={props.isEditModeActive ? 'Delete mapping' : 'Activate edit mode to delete mapping'}
+                  title={props.isEditModeActive ? t('actionmenu.delete-mapping') : t('actionmenu.activate-edit-mode-to-delete-mapping')}
                   placement="bottom"
                 >
                   <Sbutton
                     disabled={!(props.isEditModeActive)}
                     className="ms-2"
                     onClick={(e) => {
-                      props.callBackFunction.performAccordionAction(
-                        props.row,
-                        'removeMapping'
-                      );
+                      setIsDeleteMappingConfirmModalOpen(true);
                     }}
                   >
                     Delete
                   </Sbutton>
                 </Tooltip>
+                {isDeleteMappingConfirmModalOpen && <ConfirmModal
+                  heading={t('actionmenu.delete-mapping')}
+                  actionText={t('confirm')}
+                  cancelText={t('cancel')}
+                  confirmAction={performDeleteMappingAction}
+                  onClose={() => {
+                    setIsDeleteMappingConfirmModalOpen(false);
+
+                  }}
+                  heading={t('confirm-modal.heading')}
```

Comment:
Extra heading, and probably not the one you meant to show.

Reply:
Removed extra heading on next commit.